Sources/NIOCore/AsyncChannel/AsyncChannel.swift

```diff
@@ -104,6 +104,44 @@ public final class NIOAsyncChannel<Inbound: Sendable, Outbound: Sendable>: Senda
 
         self.outboundWriter.finish()
     }
+
+    @inlinable
+    @_spi(AsyncChannel)
+    public init<ChannelHandlerInboundIn: Sendable>(
+        synchronouslyWrapping channel: Channel,
+        backpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark? = nil,
+        isOutboundHalfClosureEnabled: Bool = false,
+        transformationClosure: @escaping (ChannelHandlerInboundIn) throws -> Inbound
+    ) throws where Outbound == Never {
+        channel.eventLoop.preconditionInEventLoop()
+        self.channel = channel
+        (self.inboundStream, self.outboundWriter) = try channel._syncAddAsyncHandlers(
+            backpressureStrategy: backpressureStrategy,
+            isOutboundHalfClosureEnabled: isOutboundHalfClosureEnabled,
+            transformationClosure: transformationClosure
+        )
+
+        self.outboundWriter.finish()
+    }
+
+    @inlinable
+    @_spi(AsyncChannel)
+    public init(
+        synchronouslyWrapping channel: Channel,
+        backpressureStrategy: NIOAsyncSequenceProducerBackPressureStrategies.HighLowWatermark? = nil,
+        isOutboundHalfClosureEnabled: Bool = false,
+        protocolNegotiationClosure: @escaping (Channel) -> EventLoopFuture<Inbound>
```

Comment:
I'm a bit nervous about having these two methods differ only on the label of a trailing closure, which is commonly omitted.